assignments/hackyourtemperature_W2/app.js

```diff
@@ -0,0 +1,43 @@
+import express from "express";
+import keys from "./sources/keys.js";
+
+const app = express();
+app.use(express.json());
+
+// Проверка работы сервера === test route
+app.get("/", (req, res) => {
+  res.send("hello from backend to frontend!");
+});
+
+// Получение погоды по названию города ===  get weather by city name
+app.post("/weather", async (req, res) => {
+  const cityName = req.body.cityName;
+
+  // Если город не указан === if city is not provided
+  if (!cityName) {
+    return res.status(400).json({ weatherText: "cityName is required" });
+  }
+
+  try {
+    // Запрос к OpenWeather API === request to OpenWeather API
+    const response = await fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${keys.API_KEY}&units=metric`
+    );
+    const data = await response.json();
+
+    // Если город не найден === if city is not found
+    if (data.cod !== 200) {
+      return res.status(404).json({ weatherText: "City is not found!" });
+    }

```

Comment:
This assumes that any non-200 `cod` value has to mean a 404, city not found error. It may as well be the case (I don't know this), but in case other reasons can lead to a non-200 response code, a more robust way of handling these responses would be passing the `data.cod` value to the return statement to provide help with debugging.